### PR TITLE
Jukebox Fix for #165

### PR DESCRIPTION
**Pull request by caleb-mabry**
I may have missed some functionality but I believe this covers everything to support jukebox functionality. This fixes issue #165.

The feature **CANNOT** be supported unless the length of songs is filled out appropriately. 

Please keep a close eye on this. I think I have covered everything and testing locally seems to be fine with me.

### CODE REVIEW

File: server/area_manager.py [hunk 1]
```diff
@@ -94,6 +94,10 @@ def __init__(self,
             self.is_locked = self.Locked.FREE
             self.blankposting_allowed = True
             self.non_int_pres_only = non_int_pres_only
+
+            if jukebox is True:
```
Comment (oldmud0) [inline server/area_manager.py]:
```suggestion
            if jukebox:
```

File: server/jukebox.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    def play(self):
+        song_picked, song_length = self.get_next_song_in_queue()
+        if song_picked == '':
+            self.area.broadcast_ooc('No other songs in the voting queue.')
+            return 
+
+        if self.playing is False:
+            self.area.send_command('MC', song_picked, song_length)
+            self.playing = True
+            timer_for_next_song = Timer(song_length, self.set_not_playing)
```
Comment (oldmud0) [inline server/jukebox.py]: use `asyncio.call_later`. also how does it play the next track if the timer is always hooked to `set_not_playing`?

File: server/jukebox.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            return '', -1
+
+        picked_songs = [vote.name for vote in self.votes]
+        song_frequency_occurrence = Counter(picked_songs)
+        highest_occuring: tuple = song_frequency_occurrence.most_common(1)[0]
+        highest_occuring_song = highest_occuring[0]
+
+        if len(highest_occuring) == 0:
+            highest_occuring_song = ''
+        highest_occuring_song: str = highest_occuring[0]
```
Comment (oldmud0) [inline server/jukebox.py]: missing `else`?

Comment (caleb-mabry) [reply server/jukebox.py]: You're right. Nice catch

File: server/jukebox.py (hunk 2 above)
Comment (caleb-mabry) [reply server/jukebox.py]: It sets the jukebox to `not_playing`. `set_not_playing` sets `playing` to `False` and then runs `Play` again. Then empties out the jukebox queue for the next song

Comment (caleb-mabry) [reply server/jukebox.py]: I think I described that horribly. But I did use `asyncio.call_later`

**Comment by caleb-mabry** [comment]
Closing due to me wanting to close it <3

**Pull request closed without merging**